src/components/ErrorModal.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState, useEffect } from 'react';
+import Modal from 'react-bootstrap/Modal';
+import Button from 'react-bootstrap/Button';
+import { WsCloseCodeEnum } from '../enums/wsCloseCodeEnum';
+import { provider, roomID } from '../store/store';
+
+const ErrorModal: React.FC = () => {
+	const [infoModal, setInfoModal] = useState(false);
+	const [errorReason, setErrorReason] = useState<string | null>(null);
+
+	const handleClose = () => {
+		setInfoModal(false);
+	};
+
+	const handleRedirect = () => {
+		window.location.href = `http://localhost:4000/login?redirect=tldraw?roomName=${roomID}`;
+	};
+
+	useEffect(() => {
+		const handleWsClose = (event: CloseEvent) => {
+			if (
+				[
+					WsCloseCodeEnum.WS_CLIENT_BAD_REQUEST_CODE,
+					WsCloseCodeEnum.WS_CLIENT_UNAUTHORISED_CONNECTION_CODE,
+					WsCloseCodeEnum.WS_CLIENT_ESTABLISHING_CONNECTION_CODE,
+				].includes(event.code as WsCloseCodeEnum)
+			) {
+				setErrorReason(event.reason || 'Unknown error occurred.');
+				setInfoModal(true);
+
+				if (
+					event.code === WsCloseCodeEnum.WS_CLIENT_UNAUTHORISED_CONNECTION_CODE
+				) {
+					setInfoModal(true);
+				}
+			} else {
+				setInfoModal(false);
+			}
+		};
+
+		provider.ws?.addEventListener('close', handleWsClose);
+
+		return () => {
+			provider.ws?.removeEventListener('close', handleWsClose);
+		};
+	}, []);
+
+	return (
+		<Modal show={infoModal} onHide={handleClose} centered backdrop="static">
+			<Modal.Header>
+				<Modal.Title>Error</Modal.Title>
+			</Modal.Header>
+			<Modal.Body>
+				{errorReason}
+				{errorReason &&
+					WsCloseCodeEnum.WS_CLIENT_UNAUTHORISED_CONNECTION_CODE && (
```

Comment:
This message currently looks like that:
![image](https://github.com/hpi-schul-cloud/tldraw-client/assets/27923014/e7f886ac-a24c-4079-b1bf-1097ec58d0dd)
which might be slightly misleading I guess as it might also mean that you have a stale token and you just need to log in again and not that you straight up don't have permission. Do you think it should be adjusted?